Nil ast.Expr panic with 1.13.0 in golangci-lint 2.7.1

**Describe the bug**
A clear and concise description of what the bug is.

**To Reproduce**

Running `golangci-lint run --max-issues-per-linter 0 --max-same-issues 0 --fix ./...` with `golangci-lint` 2.7.1, which seems to use revive 1.13.0, I see:

```
panic: interface conversion: ast.Expr is nil, not *ast.Ident

goroutine 1893802 [running]:
github.com/mgechev/revive/rule.(*lintInefficientMapLookup).analyzeBlock(0xc23d89a3b0, 0x1662280?)
        github.com/mgechev/revive@v1.13.0/rule/inefficient_map_lookup.go:88 +0x1d8
github.com/mgechev/revive/rule.(*lintInefficientMapLookup).Visit(0xc23d89a3b0, {0x1c1b260?, 0xc22d57b230?})
        github.com/mgechev/revive@v1.13.0/rule/inefficient_map_lookup.go:70 +0x32
go/ast.Walk({0x1c15d40?, 0xc23d89a3b0?}, {0x1c1b260, 0xc22d57b230})
        go/ast/walk.go:34 +0x4c
go/ast.Walk({0x1c15d40?, 0xc23d89a3b0?}, {0x1c1b710, 0xc23ff859c0})
        go/ast/walk.go:218 +0x2312
go/ast.walkList[...](...)
        go/ast/walk.go:21
go/ast.Walk({0x1c15d40?, 0xc23d89a3b0?}, {0x1c1b260, 0xc22d57b710})
        go/ast/walk.go:211 +0x1d6e
go/ast.Walk({0x1c15d40?, 0xc23d89a3b0?}, {0x1c1b828, 0xc2c3838000})
        go/ast/walk.go:272 +0x3307
go/ast.walkList[...](...)
        go/ast/walk.go:21
go/ast.Walk({0x1c15d40?, 0xc23d89a3b0?}, {0x1c1b260, 0xc22d57bb90})
        go/ast/walk.go:211 +0x1d6e
github.com/mgechev/revive/rule.(*InefficientMapLookupRule).Apply(0x150eb40?, 0xc0ecbd3140, {0x17720dd?, 0x16?, 0x56?})
        github.com/mgechev/revive@v1.13.0/rule/inefficient_map_lookup.go:47 +0x29b
github.com/mgechev/revive/lint.(*File).lint(0xc0ecbd3140, {0xc006141208, 0x4f, 0x8f}, {0x0, 0x3fe999999999999a, {0xc00612d05f, 0x7}, 0x1, 0x0, ...}, ...)
        github.com/mgechev/revive@v1.13.0/lint/file.go:126 +0x1e2
github.com/mgechev/revive/lint.(*Package).lint.func1()
        github.com/mgechev/revive@v1.13.0/lint/package.go:211 +0x5e
golang.org/x/sync/errgroup.(*Group).Go.func1()
        golang.org/x/sync@v0.18.0/errgroup/errgroup.go:93 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1893797
        golang.org/x/sync@v0.18.0/errgroup/errgroup.go:78 +0x95
```

I have unfortunately not been able to narrow it down to a particular piece of code that I am able to share publicly yet.

**Expected behavior**

No panic.

**Desktop (please complete the following information):**
 - Ubuntu 25.10
 - Go 1.25.3

--- a/rule/inefficient_map_lookup.go
+++ b/rule/inefficient_map_lookup.go
@@ -155,6 +155,11 @@ func (w *lintInefficientMapLookup) isRangeOverMapKey(stmt ast.Stmt) bool {
 		return false // not a range
 	}
 
+	// Check if we range on the key
+	if rangeStmt.Key == nil {
+		return false // no key in range
+	}
+
 	// Check if we range only on key
 	// for key := range ...
 	// for key, _ := range ...

--- a/testdata/inefficient_map_lookup.go
+++ b/testdata/inefficient_map_lookup.go
@@ -60,4 +60,11 @@ func inefficientMapLookup() {
 		}
 		fmt.Println(key)
 	}
+
+	// Test case for issue #1601
+	var count int
+	for range aMap {
+		count++
+	}
+	_ = count
 }

